### ISSUE TITLE
khala-node v0.1.13 getting stuck on Polkadot/Phala chain

### ISSUE THREAD

**Issue by ngrewe** [Phala-Network/phala-blockchain#782]
Heya folks,

I'm doing a bit of experimenting and am well aware that I might be veering out into "not yet supported" territory here, but I've recently tried to run v0.1.13 of khala-node against Polkadot and the Phala parachain (id 2035). I'm running into a couple of issues with that:

## Parachain blocks never get finalised

Apparently the node syncs to the head of the chain, but never marks the blocks as finalised:
```
2022-05-13 05:28:51 [Parachain] 💤 Idle (7 peers), best: #90051 (0x5d96…2488), finalized #0 (0x1bb9…d736), ⬇ 0.3kiB/s ⬆ 0.7kiB/s 
```

## Relaychain synchronisation stops

~Somewhere around Polkadot block #‍1388214, synchronisation of the relay chain just stops with the following error messages (the specific block number and block hash seem to differ if I expunge the already downloaded Polkadot chain, but it's always stalling around that approximate block height):~ (we managed to sort out that problem… sorry for the noise)

```
2022-05-13 04:01:13 [Relaychain] Block import error: Database    
2022-05-13 04:01:13 [Relaychain] 💔 Error importing block 0x679911d58df923d692ffccfbb522991577d56cf925c3106c8aa2545b002c3738: consensus error: Import failed: Import failed: Database    
2022-05-13 04:01:13 [Relaychain] 💔 Error importing block 0xd0753259965b6200e7486cc7bddc99cd77caa6ede1300fc32dc568ed773736f2: block has an unknown parent
```

Perhaps relatedly, the node reports the following errors on relaychain startup:

```
2022-05-13 03:09:25 [Relaychain] Failed to fetch runtime API data for job err=Execution { runtime_api_name: "validator_groups", source: Application(Execution(Other("Exported method ParachainHost_validator_groups is not found"))) }
2022-05-13 03:09:25 [Relaychain] Failed to fetch basics from runtime API err=Execution { runtime_api_name: "session_index_for_child", source: Application(Execution(Other("Exported method ParachainHost_session_index_for_child is not found"))) }
2022-05-13 03:09:25 [Relaychain] Our view contains 0xa172…0085 but the overseer never told use we should work on this added=0xa172…0085
2022-05-13 03:09:25 [Relaychain] job finished with an error job="bitfield-signing-job" parent_hash=0xa172…0085 err=Util(RuntimeApi(Execution { runtime_api_name: "session_index_for_child", source: Application(Execution(Other("Exported method ParachainHost_session_index_for_child is not found"))) }))
2022-05-13 03:09:25 [Relaychain] error=Sender(Runtime(RuntimeRequest(Execution { runtime_api_name: "session_index_for_child", source: Application(Execution(Other("Exported method ParachainHost_session_index_for_child is not found"))) }))) ctx="on FromOverseer"
```

## PRB cannot receive genesis state

Trying to connect the new PRB data_provider (v2.1.0) to this node, it is unable to receive the genesis state of the chain from the node:

```json
{
  "name": "prb",
  "hostname": "my-bridge",
  "pid": 23,
  "level": 30,
  "msg": "Fetching genesis...",
  "time": "2022-05-13T05:44:09.114Z",
  "v": 0
},
{
  "name": "prb",
  "hostname": "my-bridge",
  "pid": 23,
  "level": 50,
  "err": {
    "message": "Unable to retrieve header and parent from supplied hash",
    "name": "Error",
    "stack": "Error: Unable to retrieve header and parent from supplied hash\n    at assert (/opt/app/node_modules/@polkadot/util/assert.cjs:37:11)\n    at ApiPromise._getBlockRegistryViaHash (/opt/app/node_modules/@polkadot/api/base/Init.cjs:234:22)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"
  },
  "msg": "Unable to retrieve header and parent from supplied hash",
  "time": "2022-05-13T05:44:32.337Z",
  "v": 0
}
```

The commandline I'm starting the node with is the following:

```
/usr/local/bin/khala-node \
  --chain phala \
  --base-path /root/data \
  --name my-node \
  --port 30333 \
  --prometheus-port 9615 \
  --rpc-port 9933 \
  --ws-port 9944 \
  --prometheus-external \
  --ws-external \
  --rpc-cors all \
  --pruning archive --rpc-external --rpc-methods Unsafe \
  --state-cache-size 671088640 --db-cache 2048 --max-runtime-instances 100 \
  -- \
  --chain polkadot \
  --port 30334 \
  --prometheus-port 9616 \
  --rpc-port 9934 \
  --ws-port 9945 \
  --prometheus-external \
  --ws-external \
  --rpc-cors all \
  --pruning archive --rpc-external --rpc-methods Unsafe \
  --state-cache-size 671088640 --db-cache 2048 --max-runtime-instances 100
```

I'd greatly appreciate any pointers. Since there are already six nodes running on the Phala parachain, it at least seems to be somewhat possible?

Thanks!

**Comment by jasl** [comment]
Sorry for delay, this is an upstream known issue and we're waiting for the fix too.

but there's a simple way to fix just change `--state-cache-size 671088640` to `--state-cache-size 0`

BTW, `--max-runtime-instances 100` is helpless for performance, you can safely remove this

**Comment by ngrewe** [comment]
Thanks! That worked nicely for me. I found a follow-up problem, though. Cf. #799
